backend/battle/tests/test_view.py

```diff
@@ -74,3 +80,343 @@ def test_verify_if_not_returns_winner_missing_pokemons_teams(self):
+class TeamViewTest(TestCaseUtils):
+    def setUp(self):
+        super().setUp()
+        self.opponent = baker.make('users.User')
+
+    @patch('battle.views.run_battle_and_send_result_email.delay')
+    def test_if_task_run_battle_is_called(self, task_mock):
+        battle = Battle.objects.create(creator=self.user, opponent=self.opponent)
+
+        team_opponent = baker.make('battle.Team', battle=battle, trainer=self.opponent)
+
+        pokemon_1, pokemon_2, pokemon_3 = baker.make(
+            "pokemon.Pokemon",
+            attack=60, defense=45, hp=50, _quantity=3)
+
+        pokemons = [pokemon_1, pokemon_2, pokemon_3]
+
+        for count, pokemon in enumerate(pokemons):
+            PokemonTeam.objects.create(
+                team=team_opponent,
+                pokemon=pokemon,
+                order=count)
+
+        pokemons_data = {
+            "battle": battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team_create", kwargs={'pk': battle.id}), pokemons_data, follow=True)
+
+        task_mock.assert_called_with(battle.id)
+
+        self.assertTrue(Team.objects.get(battle=battle, trainer=self.user))
+
+    @patch('battle.battles.email.send_templated_mail')
+    def test_if_task_send_email(self, email_mock):
+        battle = Battle.objects.create(creator=self.user, opponent=self.opponent)
+
+        team_opponent = baker.make('battle.Team', battle=battle, trainer=self.opponent)
+
+        pokemon_1, pokemon_2, pokemon_3 = baker.make(
+            "pokemon.Pokemon",
+            attack=60, defense=45, hp=50, _quantity=3)
+
+        pokemons = [pokemon_1, pokemon_2, pokemon_3]
+
+        for count, pokemon in enumerate(pokemons):
+            PokemonTeam.objects.create(
+                team=team_opponent,
+                pokemon=pokemon,
+                order=count)
+
+        pokemons_data = {
+            "battle": battle.id,
+            "trainer": self.user.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+
+        self.auth_client.post(
+            reverse("team_create", kwargs={'pk': battle.id}), pokemons_data, follow=True)
+
+        battle_updated = Battle.objects.get(creator=self.user, opponent=self.opponent)
+
+        email_mock.assert_called_with(
+            template_name='results_battle',
+            from_email=settings.FROM_EMAIL,
+            recipient_list=[battle_updated.creator, battle_updated.opponent],
+            context={
+                'winner': battle_updated.winner,
+            },
+        )
+
+    def test_if_task_set_winner_correctly(self):
```

Comment:
This should in a separate test class since it doesn't relate to the `TeamViewTest`

Reply:
True!